src/pages/IndividualReport.vue

```diff
@@ -235,6 +235,31 @@ const exportToPdf = async () => {
     (exportLoading.value = false);
 };
 
+const optionalAssessments = computed(() =>
+  administrationData.value.assessments.filter((assessment) => assessment.optional),
+);
+
+// Calling the query client to update the cached taskData with the new optional tasks
+const queryClient = useQueryClient();
+
+const updateTaskData = () => {
+  const updatedTasks = taskData.value.map((task) => {
+    const isOptional = optionalAssessments.value.some((assessment) => assessment.taskId === task.taskId);
+    return isOptional ? { ...task, optional: true } : task;
+  });
+
+  queryClient.setQueryData(['runs', props.administrationId, props.userId], updatedTasks);
+};
+
+// Watch for changes in taskData and update the taskData with the new optional tasks
+watch(
+  () => taskData.value,
+  () => {
+    updateTaskData();
+  },
+  { deep: true },
+);
```

Comment:
Is there an advantage to using `queryClient.setQueryData` over another `computed` variable?

Reply:
I looked into this a bit, and I found that the tanstackquery has its own internal reactivity and state management considerations when attempting to directly manipulate a returned query. The computed properties that I tried to create would not effectively update the taskData, and so I had to restort to tanstackquery's interal methods.